### PR TITLE
add_test_points_to_stress

### PR DESCRIPTION
**Pull request by xiaonan-INTC**
Signed-off-by: xiaonan-INTC <xiaonan.zhao@intel.com>

### CODE REVIEW

**Comment by radhikaj** [comment]
bors try

**Comment by radhikaj** [comment]
Can you please provide some description to the PR? There were previous tests contributed to stress the memory allocator etc. Is this meant to replace those?

File: tests/stress/enc/enc.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const uint8_t* available = (const uint8_t*)dlmalloc_sbrk(0);
+    const uint8_t* end = (const uint8_t*)__oe_get_heap_end();
+    if (!(start <= available && available < end))
+        return OE_FAILURE;
+    else
+        remaining_heap = (size_t)(end - available);
+
+    // verify if heap memory is enough
+    const size_t GB = 1024 * 1024 * 1024;
+    if (!(remaining_heap > (float)(15.9 * (double)GB)))
```
Comment (radhikaj) [inline tests/stress/enc/enc.c:37]: Is the heap always expected to be greater than 16 GB?

File: tests/stress/enc/enc.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    extern void oe_free(void* ptr);
+    extern void* dlmalloc_sbrk(size_t n);
+
+    // calculate how much heap memory remains
+    size_t remaining_heap;
+    const uint8_t* start = (const uint8_t*)__oe_get_heap_base();
+    const uint8_t* available = (const uint8_t*)dlmalloc_sbrk(0);
+    const uint8_t* end = (const uint8_t*)__oe_get_heap_end();
+    if (!(start <= available && available < end))
+        return OE_FAILURE;
```
Comment (radhikaj) [inline tests/stress/enc/enc.c:31]: Is this check valid? start is an address and available is the available heap size. Is start <= available a valid check?

**Comment by radhikaj** [comment]
bors try

File: tests/stress/CMakeLists.txt [hunk 3]
```diff
@@ -8,24 +8,39 @@ if (BUILD_ENCLAVES)
 endif ()
 
 # stress test case definition:
-# add_enclave_test(tests/your_stress_test_name stress_host stress_enc stress_test_type enclave_count operation_count)
+# For create enclave & do ecall:
+#   add_enclave_test(tests/your_stress_test_name stress_host stress_enc stress_test_type enclave_count do_ecall_count)
+# For memory management:
+#   add_enclave_test(tests/your_stress_test_name stress_host stress_enc stress_test_type do_memory_count memory_size)
+# For multi-thread:
+#   add_enclave_test(tests/your_stress_test_name stress_host stress_enc stress_test_type thread_count enclave_count)
```
Comment (radhikaj) [inline tests/stress/CMakeLists.txt:16]: Does threadcount always have to match enclave count?

Comment (radhikaj) [reply tests/stress/CMakeLists.txt:16]: I think each thread creates an encalve_count number of enclaves. Closing this.

---
In reply to: [495161827](https://github.com/openenclave/openenclave/pull/3558#discussion_r495161827) [](ancestors = 495161827)

**Comment by radhikaj** [comment]
bors try

**Comment by radhikaj** [comment]
The test seems to be erroring out during the creation of the stress enclave

205/219 Test #205: tests/stress_memory_management_1 ......................................................................................................   Passed    0.17 sec
test 206
        Start 206: tests/stress_multithread_1

206: Test command: C:\Users\oeadmin\workspace\pipelines\Azure-Windows\build\tests\stress\host\stress_host.exe "C:/Users/oeadmin/workspace/pipelines/Azure-Windows/build/tests/stress/enc/stress_enc" "3" "10" "10"
206: Test timeout computed to be: 1200
206: : ..\tests\stress\host\host.cpp(110): error: oe_create_stress_enclave() - 0: result=21

**Comment by xiaonan-INTC** [comment]
I will check and refine that test point of stress test. Close this PR at present.

File: tests/stress/enc/enc.c (hunk 1 above)
Comment (xiaonan-INTC) [reply tests/stress/enc/enc.c]: refined

File: tests/stress/enc/enc.c (hunk 2 above)
Comment (xiaonan-INTC) [reply tests/stress/enc/enc.c]: refined

**Review by anakrish — APPROVED** [review]
LGTM.
Note:  The memory management stress test makes 10000 ecalls instead of making just a single ecall and then doing all the allocations in that ecall. Is there a reason why the former is preferred over the latter for this stress test?

**Comment by anakrish** [comment]
bors try

**Comment by xiaonan-INTC** [comment]
> ## try
> Build failed:
> 
> * [continuous-integration/jenkins/branch](https://oe-jenkins-dev.westeurope.cloudapp.azure.com/job/Bors/job/trying/1213/display/redirect)

@anakrish 

Dear anakrish, do you know the build failure https://oe-jenkins-dev.westeurope.cloudapp.azure.com/blue/organizations/jenkins/Bors/detail/trying/1213/
I cannot find the root cause in that web page... :(

**Comment by radhikaj** [comment]
bors try

**Comment by radhikaj** [comment]
bors try

**Comment by dthaler** [comment]
@xiaonan-INTC Can you rebase?

**Comment by radhikaj** [comment]
@xiaonan-INTC, I will close this for now. Please reopen whenever you would like to address comments.

**Pull request closed without merging**